src/brains2/src/sim/sim.cpp

```diff
@@ -0,0 +1,239 @@
+            return tl::make_unexpected(SimError::SAMPLING_TIME_UPDATE_ERROR);
+        }
+
+        // Call simulation solver
+        exit_code = dyn6_acados_sim_solve((dyn6_sim_solver_capsule *)dyn6_workspace._sim_capsule);
+        if (exit_code != 0) {
+            return tl::make_unexpected(SimError::ACADOS_SOLVER_ERROR);
+        }
+
+        // Get next state
+        sim_out_get(dyn6_workspace._sim_config,
+                    dyn6_workspace._sim_dims,
+                    dyn6_workspace._sim_out,
+                    "xn",
+                    x_next.data());
+
+        exit_code = casadi_eval(dyn6_workspace.accel_fun_mem);
+        if (exit_code != 0) {
+            return tl::make_unexpected(SimError::ACCELS_FUNCTION_ERROR);
+        }
+    } else {
+        // We use the kinematic model
+        // Set simulation solver inputs
+        sim_in_set(kin6_workspace._sim_config,
+                   kin6_workspace._sim_dims,
+                   kin6_workspace._sim_in,
+                   "x",
+                   x.data());
+        sim_in_set(kin6_workspace._sim_config,
+                   kin6_workspace._sim_dims,
+                   kin6_workspace._sim_in,
+                   "u",
+                   u.data());
+
+        int exit_code = sim_in_set(kin6_workspace._sim_config,
+                                   kin6_workspace._sim_dims,
+                                   kin6_workspace._sim_in,
+                                   "T",
+                                   &dt);
+        if (exit_code != 0) {
+            return tl::make_unexpected(SimError::SAMPLING_TIME_UPDATE_ERROR);
+        }
+
+        // Call simulation solver
+        exit_code = kin6_acados_sim_solve((kin6_sim_solver_capsule *)kin6_workspace._sim_capsule);
+        if (exit_code != 0) {
+            return tl::make_unexpected(SimError::ACADOS_SOLVER_ERROR);
+        }
+
+        // Get next state
+        sim_out_get(kin6_workspace._sim_config,
+                    kin6_workspace._sim_dims,
+                    kin6_workspace._sim_out,
+                    "xn",
+                    x_next.data());
+
+        exit_code = casadi_eval(kin6_workspace.accel_fun_mem);
+        if (exit_code != 0) {
+            return tl::make_unexpected(SimError::ACCELS_FUNCTION_ERROR);
+        }
+    }
+
+    State next_state{x_next[0],
+                     x_next[1],
+                     x_next[2],
+                     x_next[3],
+                     x_next[4],
+                     x_next[5],
+                     x_next[6],
+                     x_next[7],
+                     x_next[8],
+                     x_next[9],
+                     x_next[10]};
+    Accels next_accels{a[0], a[1]};
+
+    // Check for NaNs
+    if (std::any_of(x_next.begin(), x_next.end(), [](double value) { return std::isnan(value); }) ||
+        std::any_of(a.begin(), a.end(), [](double value) { return std::isnan(value); })) {
+        return tl::make_unexpected(SimError::NANS_IN_RESULT);
+    }
```

Comment:
🫨🫨🫨

Reply:
behold the power of modern C++ 
( not sure why I couldn't just `std::any_of(x_next.begin(), x_next.end(), std::isnan)` tho...)

Reply:
We leave it like that then ?